### PR TITLE
Add reclaim function

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

File: contracts/truefi/LoanToken.sol [hunk 1]
```diff
@@ -305,6 +312,20 @@ contract LoanToken is ILoanToken, ERC20 {
         emit Repaid(_sender, _amount);
     }
 
+    /**
+     * @dev Function for borrower to reclaim stuck currencyToken
+     * Can only call this function after the loan is Closed
+     * and all of LoanToken holders have fully redeemed
```
Comment (vanruch) [inline contracts/truefi/LoanToken.sol:318]:
```suggestion
     * and all of LoanToken holders have been burnt
```

File: contracts/truefi/LoanToken.sol [hunk 2]
```diff
@@ -305,6 +312,20 @@ contract LoanToken is ILoanToken, ERC20 {
         emit Repaid(_sender, _amount);
     }
 
+    /**
+     * @dev Function for borrower to reclaim stuck currencyToken
+     * Can only call this function after the loan is Closed
+     * and all of LoanToken holders have been burnt
+     */
+    function reclaim() external override onlyClosed onlyBorrower {
+        require(totalSupply() == 0, "LoanToken: Cannot reclaim when LoanTokens in circulation");
```
Comment (barrutko) [inline contracts/truefi/LoanToken.sol:321]:
```suggestion
        require(totalSupply() == 0, "LoanToken: Cannot reclaim when LoanTokens are in circulation");
```

File: test/truefi/LoanToken.test.ts [hunk 3]
```diff
@@ -384,6 +384,80 @@ describe('LoanToken', () => {
...
+    it('reverts when not borrower tries access', async () => {
+      await loanToken.close()
+      await expect(loanToken.reclaim())
+        .to.be.revertedWith('LoanToken: Caller is not the borrower')
+    })
+
+    it('reverts when total supply is greater than 0', async () => {
+      await loanToken.close()
+      await expect(loanToken.connect(borrower).reclaim())
+        .to.be.revertedWith('LoanToken: Cannot reclaim when LoanTokens in circulation')
```
Comment (barrutko) [inline test/truefi/LoanToken.test.ts:409]:
```suggestion
        .to.be.revertedWith('LoanToken: Cannot reclaim when LoanTokens are in circulation')
```

File: test/truefi/LoanToken.test.ts [hunk 4]
```diff
@@ -384,6 +384,80 @@ describe('LoanToken', () => {
...
+      await loanToken.redeem(parseEther('1100'))
+      await expect(loanToken.connect(borrower).reclaim())
+        .to.be.revertedWith('LoanToken: Cannot reclaim when balance 0')
+    })
+
+    const paybackRedeemPayback = async () => {
+      await payback(borrower, parseEther('900'))
+      await loanToken.redeem(parseEther('1100'))
+      await payback(borrower, parseEther('200'))
+    }
```
Comment (barrutko) [inline test/truefi/LoanToken.test.ts:424]: Helpers definitions should be located at the beginning of `describe`, above all all `it`s

**Review by barrutko — COMMENTED** [review]
@vanruch  can have a final look, but imo looks generally good

File: test/truefi/LoanToken.test.ts (hunk 4 above)
Comment (MiksuJak) [reply test/truefi/LoanToken.test.ts]: Done

**Review by vanruch — APPROVED** [review]

**Pull request merged**